src/libasr/codegen/asr_to_c.cpp

```diff
@@ -645,6 +645,32 @@ R"(
             array_types_decls += src;
         }
 
+        if (x.m_global_scope->get_symbol("_global_symbols") != nullptr) {
+            struct_dep_graph.clear();
+            SymbolTable *global_symbols = ASR::down_cast<ASR::Module_t>(
+                x.m_global_scope->get_symbol("_global_symbols"))->m_symtab;
+            for (auto &item : global_symbols->get_scope()) {
+                if (ASR::is_a<ASR::StructType_t>(*item.second) ||
+                        ASR::is_a<ASR::EnumType_t>(*item.second) ||
+                        ASR::is_a<ASR::UnionType_t>(*item.second)) {
+                    std::vector<std::string> struct_deps_vec;
+                    std::pair<char**, size_t> struct_deps_ptr = ASRUtils::symbol_dependencies(item.second);
+                    for( size_t i = 0; i < struct_deps_ptr.second; i++ ) {
+                        struct_deps_vec.push_back(std::string(struct_deps_ptr.first[i]));
+                    }
+                    struct_dep_graph[item.first] = struct_deps_vec;
+                }
+            }
+
+            std::vector<std::string> struct_deps = ASRUtils::order_deps(struct_dep_graph);
+
+            for (auto &item : struct_deps) {
+                ASR::symbol_t* struct_sym = global_symbols->get_symbol(item);
+                visit_symbol(*struct_sym);
+                array_types_decls += src;
+            }
+        }
```

Comment:
Why do we need special handling of this global_symbols module? I was hoping that since it's just a regular module, it would not need any special handling like this.

Reply:
Yup, here in C backend, I think the visit_module is not yet implemented, except `lfortran_intrinsics` module.

Reply:
Correct. Probably this can be removed. I guess you can just do `if(ASR::is_a<ASR::Module_t>(*item.second))` and call `visit_Module(*ASR::down_cast<ASR::Module_t>(item.second))` inside the `if` block by creating a new loop similar to the following,

https://github.com/lcompilers/lpython/blob/8f62ac4ff1acebdeb621557d8008a5a8423455bb/src/libasr/codegen/asr_to_c.cpp#L613-L623

Please comment here if you face any blocker or issues while following this approach.